### PR TITLE
feat: Shift Registers for Shift+Trigger operations

### PR DESCRIPTION
**Pull request by jstarpl**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Feature

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

You can set up actions that modify an internal, global set of "Shift Registers" in the Input Gateway. The "Shift registers" state modifies the triggers emitted from Input Gateway by prefixing the trigger string with a string describing the Shift Registers state, effectively creating a new trigger. This allows for creating "Shift + Key" operations, latching "modes", pagination and cascading menus.

* **Other information**:

### CODE REVIEW

File: packages/input-manager/src/devices/feedbackStore.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { SomeFeedback } from '../feedback/feedback'
+
+export class FeedbackStore<T extends SomeFeedback> {
```
Comment (nytamin) [inline packages/input-manager/src/devices/feedbackStore.ts:3]: Question: You're not really using this class as a generic anywhere, perhaps just use `SomeFeedback` directly?

File: packages/input-manager/src/integrations/streamdeck/index.ts [hunk 2]
```diff
@@ -91,10 +111,13 @@ export class StreamDeckDevice extends Device {
 				if (!prev) prev = { deltaValue: 0 }
 				return {
 					deltaValue: prev.deltaValue - deltaValue,
+					direction: -1,
 				}
 			})
 
-			this.updateFeedback(id, false).catch((err) => this.logger.error(`Stream Deck: Error updating feedback: ${err}`))
+			this.updateFeedback(id, this.#isButtonDown[id] ?? false).catch((err) =>
+				this.logger.error(`Stream Deck: Error updating feedback: ${err}`)
```
Comment (nytamin) [inline packages/input-manager/src/integrations/streamdeck/index.ts:119]: There are lots of these, perhaps refactor into a single error-handling function? or possibly change `updateFeedback` to not be an async function that handles it's own errors?

File: packages/input-manager/src/devices/feedbackStore.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		}
+
+		return null
+	}
+
+	public clear(): void {
+		this.feedbacks = {}
+	}
+
+	public allFeedbacks(): string[] {
```
Comment (nytamin) [inline packages/input-manager/src/devices/feedbackStore.ts:32]:
```suggestion
	public allFeedbackIds(): string[] {
```

**Review by nytamin — APPROVED** [review]
Great work!
This will be a very powerful and generic addition!

One note that I have is that there should probably be some written documentation detailing the overall shiftRegister functionality, possibly along with some basic examples (what actually happens when shift+S is pressed, how a page is set up etc).

File: packages/input-manager/src/devices/feedbackStore.ts (hunk 1 above)
Comment (jstarpl) [reply packages/input-manager/src/devices/feedbackStore.ts:3]: Well, I'm not using it right now, but the plan was that the FeedbackStore could be used for objects enhancing the SomeFeedback interface, that's why it's generic.

**Pull request merged**